dlib/image_loader/jxl_loader_abstract.h

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2024  Davis E. King (davis@dlib.net), Adrià Arrufat
+// License: Boost Software License   See LICENSE.txt for the full license.
+#undef DLIB_JXL_IMPORT_ABSTRACT
+#ifdef DLIB_JXL_IMPORT_ABSTRACT
+
+#include "image_loader_abstract.h"
+#include "../algs.h"
+#include "../pixel.h"
+#include "../dir_nav.h"
+#include "../image_processing/generic_image.h"
+
+namespace dlib
+{
+
+    class jxl_loader : noncopyable
```

Comment:
How about only having this class in the .cpp file?  The png and jpeg ones have other methods that let you quickly query stuff about the image.  But if there is only the get_image() method then there isn't any reason for the class to be public to users, and they are better off just always using the global methods.

Reply:
OK, I'll try that later.

Reply:
I've ended up adding those methods to the `jxl_loader`.
I didn't know dlib had those.

Furthermore, I took the liberty to add `nr()` and `nc()` to query the height and width, respectively. With these, I could even discard images with undesirable sizes before fully loading them.

Maybe `width()` and `height()` would be better for method names? I just used `nr` and `nc` to be consistent with how the ones in the `matrix` class are named.

Reply:
Yeah `nr()` and `nc()` is good.